src/reducers/chat.js

```diff
@@ -1,12 +1,22 @@
+import isEmpty from 'lodash/isEmpty';
+import findIndex from 'lodash/findIndex';
+import assign from 'lodash/assign';
 import {
   GET_ALL_MESSAGES,
   CHAT_ADD_MESSAGE,
   CHAT_MESSAGE_IS_TYPING,
-  // CHAT_REMOVE_MESSAGE,
+  CHAT_REMOVE_MESSAGE,
+  CHAT_UPDATE_MESSAGE_BY_ID,
+  EDIT_MESSAGE,
 } from '../constants';
 
 export default function chat(state = { messages: [] }, action) {
   const { messages } = state;
+  const editMessageIndex = findIndex(messages, { id: action.messageId });
+  const newArrayMessages = Array.from(
+    assign(messages, (messages[editMessageIndex] = action.message)),
```

Comment:
You start to make me hate you :-) please don't make assignments inside a function call

Reply:
Please show me have to do it right, because 
I don't know how else to do than I did.